pkg/scheduler/api/job_info.go

```diff
@@ -349,3 +349,57 @@ func (ji *JobInfo) FitError() string {
 	reasonMsg := fmt.Sprintf("0/%v nodes are available, %v.", len(ji.NodesFitDelta), strings.Join(sortReasonsHistogram(), ", "))
 	return reasonMsg
 }
+
+// readyTaskNumForRun return the number of tasks that are ready to run.
+func (ji *JobInfo) ReadyTaskNumForRun() int32 {
+	occupid := 0
+	for status, tasks := range ji.TaskStatusIndex {
+		if AllocatedStatus(status) ||
+			status == Succeeded {
+			occupid = occupid + len(tasks)
+		}
+	}
+
+	return int32(occupid)
+}
+
+// readyTaskNumForRun return the number of tasks that are ready for preempt.
+func (ji *JobInfo) ReadyTaskNumForPreempt() int32 {
```

Comment:
prefer to `WaitingTaskNum` which only includes `Pipelined`; the `Releasing` resource maybe triggerred by reclaim.